solidity-v1/dashboard/src/pages/threshold/ThresholdUpgradePage.jsx

```diff
@@ -0,0 +1,316 @@
+  }, [isDelegationDataFetching, isGrantDataFetching])
+
+  const { totalOwnedStakedBalance, totalOwnedUnstakedBalance } =
+    useKeepBalanceInfo()
+
+  const { totalGrantedStakedBalance, totalGrantedUnstakedBalance } =
+    useGrantedBalanceInfo()
+
+  const notStakedTotalAmount = useMemo(() => {
+    return add(
+      totalOwnedUnstakedBalance,
+      totalGrantedUnstakedBalance
+    ).toString()
+  }, [totalOwnedUnstakedBalance, totalGrantedUnstakedBalance])
+
+  const stakedTotalAmount = useMemo(() => {
+    return add(totalOwnedStakedBalance, totalGrantedStakedBalance).toString()
+  }, [totalOwnedStakedBalance, totalGrantedStakedBalance])
+
+  const {
+    totalStakedPendingKeep,
+    totalStakedAvailableKeep,
+    totalUndelegatedAvailableKeep,
+  } = useMemo(() => {
+    const totalStakedPendingKeep = [...undelegations]
+      .filter((delegation) => !delegation.canRecoverStake)
+      .map(({ amount }) => amount)
+      .reduce(add, "0")
+      .toString()
+
+    const totalStakedAvailableKeep = [...delegations]
+      .map(({ amount }) => amount)
+      .reduce(add, "0")
+      .toString()
+
+    const totalUndelegatedAvailableKeep = [...undelegations]
+      .filter((delegation) => delegation.canRecoverStake)
+      .map(({ amount }) => amount)
+      .reduce(add, "0")
+      .toString()
+
+    return {
+      totalStakedPendingKeep,
+      totalStakedAvailableKeep,
+      totalUndelegatedAvailableKeep,
+    }
+  }, [delegations, undelegations])
+
+  const onWithdrawFromGrant = () => {
+    console.log("withdraw from grant clicked!")
+    openModal(MODAL_TYPES.WithdrawGrantedTokens, { grants })
+  }
+
+  return (
+    <section className="threshold-upgrade-page">
+      <section className="tile threshold-upgrade-page__explanation">
+        <Icons.KeepTUpgrade className="threshold-upgrade-page__explanation__icon" />
+        <header>
+          <h2 className="text-grey-70 threshold-how-it-works-page__explanation-title">
+            Upgrade Your KEEP to T
+          </h2>
+          <h3 className="text-grey-50 threshold-how-it-works-page__explanation-description">
+            Threshold Network is the network merger between Keep and NuCypher.
+            Upgrade your KEEP to T below.
+          </h3>
+        </header>
+        <NavLink
+          to="/threshold/how-it-works"
+          className="btn btn-secondary btn-md explanation__upgrade-btn"
+        >
+          learn more
+        </NavLink>
+      </section>
+
+      <section className="tile not-staked">
+        <h3 className="mb-1">Not staked</h3>
+        <OnlyIf condition={isDataFetching}>
+          <div className="not-staked__token-amount not-staked__token-amount--loading">
+            <Icons.KeepOutline
+              width={32}
+              height={32}
+              className={`token-amount__icon`}
+            />
+            <Skeleton tag="h2" shining color="grey-20" width="30%" />
+          </div>
+        </OnlyIf>
```

Comment:
Same as above. And maybe it will be easier to read if we use `ternary operator`? `isDataFetching ? <TokenAmountSkeleton /> : <TokenAmount {...} />`

Reply:
Fixed in e3d3a0a44fba2c556ed2ffa14ed7d5b5819534c6